src/context/authentication-context.js

```diff
@@ -0,0 +1,58 @@
+import { createContext, useContext, useReducer } from "react";
+
+const AuthenticationContext = createContext();
+const useAuthenticationContext = () => useContext(AuthenticationContext);
+
+const authenticationReducer = (state, { type, payload }) => {
+  switch (type) {
+    case "USER":
+      return { ...state, user: payload };
+    case "EMAIL":
+      return { ...state, email: payload };
+    case "PASSWORD":
+      return { ...state, password: payload };
+    case "CONFIRM_PASSWORD":
+      return { ...state, confirmPassword: payload };
+    case "FIRST_NAME":
+      return { ...state, firstName: payload };
+    case "LAST_NAME":
+      return { ...state, lastName: payload };
+    case "CLEAR_SIGN_UP":
+      return {
+        ...state,
+        firstName: "",
+        lastName: "",
+        email: "",
+        password: "",
+        confirmPassword: "",
+      };
+    case "CLEAR_SIGN_IN":
+      return {
+        ...state,
+        email: "",
+        password: "",
+      };
+    default:
+      return state;
+  }
+};
+
+const AuthenticationProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authenticationReducer, {
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+    user: {},
+  });
+
+  const value = { authState, authDispatch };
```

Comment:
suggestion: you can directly add the values no need to create variable.

Reply:
I think this way is more readable if we have to send more variables and functions to context.